model/riscv_insts_zce_ext.sail

```diff
@@ -0,0 +1,1021 @@
+/*  2. Redistributions in binary form must reproduce the above copyright                 */
+/*     notice, this list of conditions and the following disclaimer in                   */
+/*     the documentation and/or other materials provided with the                        */
+/*     distribution.                                                                     */
+/*                                                                                       */
+/*  THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS''                   */
+/*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED                    */
+/*  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A                      */
+/*  PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR                  */
+/*  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,                         */
+/*  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT                     */
+/*  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF                     */
+/*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND                  */
+/*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,                   */
+/*  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT                   */
+/*  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF                   */
+/*  SUCH DAMAGE.                                                                         */
+/*=======================================================================================*/
+
+/* ****************************************************************** */
+/* This file specifies the instructions in the 'ZCEE' extension.      */
+/* ****************************************************************** */
+
+/* ------------------------------------------------------------------ */
+/* Sail Model for the c.l and c.s encoding group                */
+/* ------------------------------------------------------------------ */
+
+/* ****************************************************************** */
+union clause ast = C_LBU : (bits(2), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ C0]
+mapping clause encdec_compressed = C_LBU(ui65, rs1, rd)
+  <-> 0b100 @ 0b000 @ rs1 : cregidx @ ui65 : bits(2) @ rd : cregidx @ 0b00
+
+function clause execute (C_LBU(uimm, rsc, rdc)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+
+  let rd = creg2reg_idx(rdc);
+  let rs = creg2reg_idx(rsc);
+
+  let rsext = X(rs) + immext;
+  X(rs) = rsext;
+
+   match execute(LOAD(imm, rs, rd, false, WORD, false, false)) {
+     RETIRE_SUCCESS => {
+                         X(rd) = EXTZ(X(rd)[7..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
```

Comment:
This is clearly nonsense

Reply:
Thank you for the comments.

I was recommended to adopt this code, after i saw some issues while compiling the Sail code. Could you please let me why this is nonsense, so, i would be able to understand my mistake and also get by better in Sail.

Also, regarding the previous comment on the name for the Zce, i have a separate extension document with Zc as the the document name, so, used Zce.

Reply:
@Alasdair I believe coding this "nonsense" was your recommendation?

Reply:
I highly doubt he recommended implementing "load byte unsigned" as "load word and mask the top bits off"